### ISSUE TITLE
Support KSP

### ISSUE THREAD

**Issue by ddk9499** [stephanenicolas/toothpick#424]
For Kotlin projects, Dagger currently makes use of KAPT. This is generally quite slow since it requires generating Java stubs before the annotation processing step.

[KSP](https://github.com/google/ksp) offers an alternative to this by making it easier for annotation processors like Dagger to more directly use Kotlin's compiler plugin infrastructure. Based on initial benchmarks, using KSP can speed up build times by as much as 2x.

**Comment by L7ColWinters** [comment]
I don't see any pr's about this currently. Has anyone taken a try at this yet? It's been about a year and other libraries have added ksp adaptations

**Comment by afaucogney** [comment]
@L7ColWinters it's smell like @outadoc is working on this : https://github.com/stephanenicolas/toothpick/pull/433

**Comment by outadoc** [comment]
Indeed! I've had little time to work on it so far so I can't make any guarantees on stability or compatibility, and there are KSP issues that might need fixes for all the tests to pass, but it's something, and you should be able to run it on a real project.

If anyone feels like testing and fixing stuff, contributions are welcome: https://github.com/outadoc/toothpick-compiler-ksp
